neo/rawio/axonarawio.py

```diff
@@ -0,0 +1,374 @@
+        sig_ids[0::3] = sample1
+        sig_ids[1::3] = sample2
+        sig_ids[2::3] = sample3
+        sig_ids = sig_ids[rem:(rem+num_samples)]
+
+        # Read one channel at a time
+        raw_signals = np.ndarray(shape=(num_samples, len(channel_indexes)))
+
+        for i, ch_idx in enumerate(channel_indexes):
+
+            chan_offset = self.get_channel_offset(ch_idx)
+            raw_signals[:,i] = self._raw_signals[sig_ids + chan_offset]
+
+        return raw_signals
+
+    def _spike_count(self, block_index, seg_index, unit_index):
+        return None
+
+    def _get_spike_timestamps(self, block_index, seg_index, unit_index, t_start, t_stop):
+        return None
+
+    def _rescale_spike_timestamp(self, spike_timestamps, dtype):
+        return None
+
+    def _get_spike_raw_waveforms(self, block_index, seg_index, unit_index, t_start, t_stop):
+        return None
+
+    def _event_count(self, block_index, seg_index, event_channel_index):
+        return None
+
+    def _get_event_timestamps(self, block_index, seg_index, event_channel_index, t_start, t_stop):
+        return None
+
+    def _rescale_event_timestamp(self, event_timestamps, dtype):
+        return None
+
+    def _rescale_epoch_duration(self, raw_duration, dtype):
+        return None
+
+    # ------------------ HELPER METHODS --------------------
+    # These are credited largely to Geoff Barrett from the Hussaini lab:
+    # https://github.com/GeoffBarrett/BinConverter
+    # Adapted or modified by Steffen Buergers
+
+    def get_active_tetrode(self):
+        """ 
+        Returns the ID numbers of the active tetrodes (those with recorded data)
+        as a list. E.g.: [1,2,3,4] for a recording with 4 tetrodes (16 channels).
+        """
+        active_tetrodes = []
+
+        with open(self.set_file, encoding=self.set_file_encoding) as f:
+            for line in f:
+
+                # The pattern to look for is collectMask_X Y, 
+                # where X is the tetrode number, and Y is true or false (1, 0)
+                if 'collectMask_' in line:
+                    tetrode_str, tetrode_status = line.split(' ')
+                    if int(tetrode_status) == 1:
+                        tetrode_id = int(re.findall(r'\d+', tetrode_str)[0])
+                        active_tetrodes.append(tetrode_id)
+
+        return active_tetrodes
+
+    def get_channel_from_tetrode(self, tetrode):
+        """ 
+        This function will take the tetrode number and return the Axona channel numbers
+        i.e. Tetrode 1 = Ch1-Ch4, Tetrode 2 = Ch5-Ch8, etc.
+        """
+        return np.arange(1, 5) + 4 * (int(tetrode) - 1)
+
+    def get_channel_offset(self, chan_id):
+        """ 
+        In the .bin file, channels are arranged in a strange order. This method takes
+        a channel index as input and returns the actual offset for the channel in the
+        memory map (self._raw_signals).
+        """
+        channnel_memory_offset = np.array(
```

Comment:
I would put this variable as global because it is instancieted at each get_analogsignal_chunk and will maybe have some impact of perf.

Reply:
I added it as class variable. Is that good practice?